contracts/V1/StateChannelDiamondProxy/DisputeManagerFacet.sol

```diff
@@ -553,8 +521,56 @@ contract DisputeManagerFacet is StateChannelCommon {
         }
         return previousJoinChannelBlockHash == disputeAuditingData.outputStateSnapshot.latestExitChannelBlockHash;
 
+    }   
+    
+    function _verifyJoinChannelBlocks(Dispute memory dispute, DisputeAuditingData memory disputeAuditingData) internal pure returns (bool) {
+        //check joinChannelBlocks (linked to latestSateSnapshot, chained internally and outputStateSnapshot commits to the head)
+        bytes32 previousJoinChannelBlockHash = disputeAuditingData.latestStateSnapshot.latestJoinChannelBlockHash;
+        for(uint i = 0; i < disputeAuditingData.joinChannelBlocks.length; i++) {
+            if(previousJoinChannelBlockHash != disputeAuditingData.joinChannelBlocks[i].previousBlockHash) {
+                return false;
+            }
+            previousJoinChannelBlockHash = keccak256(abi.encode(disputeAuditingData.joinChannelBlocks[i]));
+        }
+        return previousJoinChannelBlockHash == disputeAuditingData.outputStateSnapshot.latestExitChannelBlockHash;
```

Comment:
why is it checking on existBlockChannelBlockHash on return

Reply:
fixed this